cli_helpers/tabular_output/delimited_output_adapter.py

```diff
@@ -27,10 +32,15 @@ def adapter(data, headers, table_format='csv', **kwargs):
     ckwargs = {'delimiter': delimiter}
     ckwargs.update(filter_dict_by_key(kwargs, keys))
 
-    with contextlib.closing(StringIO()) as content:
-        writer = csv.writer(content, **ckwargs)
-        writer.writerow(headers)
-        for row in data:
-            writer.writerow(row)
-
-        return content.getvalue()
+    l = linewriter()
+    writer = csv.writer(l, **ckwargs)
+    l.reset()
+    writer.writerow(headers)
+    for line in l.lines:
```

Comment:
I don't think that's possible without leaving a loop in adapter (yield is bound to the function/method as far as I know). I've removed the loop; I assume every writerow directly calls our write function (with all the data we've supplied).